### ISSUE TITLE
s3 error and autoscaling not starting

### ISSUE THREAD

**Issue by tdeheurles** [kz8s/tack#15]
Hi, first thanks for project, I learned a lot from it.

I'm trying to start your kube cluster.

![image](https://cloud.githubusercontent.com/assets/9496645/15455912/81e3db82-2061-11e6-95a3-af89af31bb2d.png)

I have tried the deployment through a docker container first and now with an ubuntu vm. I have always the same issue. Do you have an idea on that s3 problem ? I'm admin with the aws account used.

**Comment by tdeheurles** [comment]
Maybe it could help:

After a few moment, I have tried to run `make clean`. The cluster was finally started and the autoscaling was created. `make clean` was failing to delete everything, blocking on the subnet deletion. 
After I manually removed the scaling group, the `make clean` command ran successfully.

**Comment by wellsie** [comment]
Hey Thibault, you are welcome ;)

Is it working for you now ? Lmk if it is and I'll dig into that code.

**Comment by tdeheurles** [comment]
I will test again today. Maybe the cluster was finally up but the kubeconfig was not generated and kubectl is not configured.

~~What are these s3 push for ?~~
I suppose that's the way you share the credentials between the nodes.

If you are interested, I have a script to generate and run the container and also one to install the dependencies for ubuntu.

**Comment by tdeheurles** [comment]
I'm looking to [that file](https://github.com/kz8s/tack/blob/cda8fb8ba27ba4ae24ec0abfff4a200839c441cc/modules/s3/s3-cp#L49).
The `$HYPERKUBE` initialisation seems commented. Did you forgotten something when updating to 1.2.4 ?

**Comment by tdeheurles** [comment]
### s3 cp

s3 cp issue is linked to aws configuration.

At first, my `aws cli` was configured to `eu-central-1` and the region for the tack deployment was `us-west-1`. I switched my `aws cli` configuration to `us-west-1` and that s3 cp problem gone.

Maybe we could add the region explicitly [here](https://github.com/kz8s/tack/blob/master/modules/s3/s3.tf#L12) as an ENV ?

### autoscaling

The autoscaling still timeout.

Here are the outputs:

```
module.worker.aws_autoscaling_group.worker: Still creating... (9m50s elapsed)
module.worker.aws_autoscaling_group.worker: Still creating... (10m0s elapsed)
Error applying plan:

1 error(s) occurred:

* aws_autoscaling_group.worker: "worker": Waiting up to 10m0s: Need at least 5 healthy instances in ASG, have 3

Terraform does not automatically rollback in the face of errors.
Instead, your Terraform state file has been partially updated with
any resources that successfully completed. Please address the error
above and apply again to incrementally change your infrastructure.
makefiles/terraform.mk:11: recipe for target 'apply' failed
make: *** [apply] Error 1
➜  tack git:(master) ✗ make output kubeconfig
make: *** No rule to make target 'output'.  Stop.
➜  tack git:(master) ✗ terraform output kubeconfig
The output variable requested could not be found in the state
file. If you recently added this to your configuration, be
sure to run `terraform apply`, since the state won't be updated
with new output variables until that command is run.
➜  tack git:(master) ✗ kubectl get nodes
NAME                                       STATUS    AGE
ip-10-0-10-10.us-west-1.compute.internal   Ready     36m
ip-10-0-10-9.us-west-1.compute.internal    Ready     36m
ip-10-0-11-86.us-west-1.compute.internal   Ready     36m
➜  tack git:(master) ✗ 
```

I will let the cluster run to see if it finally scale. If it's so long, maybe it shouldn't fail for that ? Or maybe that scaling is abnormally slow ?

### kubectl configuration

kubectl was configured this time.
Note that `terraform output kubeconfig` didn't work for me:

```
➜  tack git:(master) ✗ terraform output kubeconfig
The output variable requested could not be found in the state
file. If you recently added this to your configuration, be
sure to run `terraform apply`, since the state won't be updated
with new output variables until that command is run.
```

I suppose it's a problem linked to the fact terraform didn't finished sucessfully (due to the timeout).

**Comment by tdeheurles** [comment]
autoscaling after 3 hours:
![image](https://cloud.githubusercontent.com/assets/9496645/15468443/93af0ce6-20e4-11e6-8062-4ff793167b41.png)

**Comment by wellsie** [comment]
The problem with the asg is most definitely due to payloads not copying to the s3 bucket - as per the screenshot. Although I wasn't able to reproduce the issue myself, I suspect it is due to the _ambient_ region from your environment not matching the region used by terraform. The error message does say to pass the `--region` flag to `aws s3 cp`.

I have updated the code to pass in the region to the script that sends the assets to s3. I've tested it in `us-west-1` and `us-east-1` is running now. Watch for a commit shortly with the fix. 

@tdeheurles lmk if it works for you.
